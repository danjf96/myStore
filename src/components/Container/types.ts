import React from "react";
import { ViewProps, ViewStyle } from "react-native";

export interface PROPSCONTAINER extends ViewProps {
    style?: ViewStyle
}