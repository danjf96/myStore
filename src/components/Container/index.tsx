
import React from 'react'
import { SafeAreaView, View } from 'react-native'
import Styles from './styles'
import { PROPSCONTAINER } from './types'

const Container = (props:PROPSCONTAINER) => {
    return (
        <SafeAreaView>
            <View {...props} style={{ ...Styles.container, ...props.style}} />
        </SafeAreaView>
    )
}

export default Container