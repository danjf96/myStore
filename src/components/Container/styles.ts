import { Platform, StyleSheet } from "react-native";

const Styles = StyleSheet.create({
    container: {
        width:'100%',
        padding: 15
    }
})

export default Styles