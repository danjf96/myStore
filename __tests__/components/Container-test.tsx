/**
 * @format
 */

import 'react-native';
import React from 'react';
import Container from '../../src/components/Container';
import { render } from 'react-native-testing-library';

describe('Render component', () => {
    it('renders correctly', async () => { 
        const screen = render( <Container />)
        expect(screen.toJSON()).toMatchSnapshot()
    });

})