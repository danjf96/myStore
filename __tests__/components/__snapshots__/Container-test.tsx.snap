// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "padding": 15,
        "width": "100%",
      }
    }
  />
</RCTSafeAreaView>
`;
